### PR TITLE
WIP: Change Field to bsiGroup

### PR DESCRIPTION
**Pull request by travisturner**
This commit renames Field to `BSIGroup` (i.e. it doesn't unexport anything).
A following commit should unexport `BSIGroup` and change frames to
support a frame type of `int` which will implement `bsiGroup`.

## Overview

Partially addresses #1314.

## Pull request checklist

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [x] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [x] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [x] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -488,9 +488,9 @@ func (api *API) Schema(ctx context.Context) []*IndexInfo {
 	return api.Holder.Schema()
 }
 
-// CreateField creates a new BSI field in the given index and frame.
-func (api *API) CreateField(ctx context.Context, indexName string, frameName string, field *Field) error {
-	if err := api.validate(apiCreateField); err != nil {
+// CreateBSIGroup creates a new BSI bsiGroup in the given index and frame.
```
Comment (jaffee) [inline api.go]: comment is awkward now, but we can fix that later

File: cmd/import.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ omitted. If it is present then its format should be YYYY-MM-DDTHH:MM.
 	flags.StringVarP(&Importer.Host, "host", "", "localhost:10101", "host:port of Pilosa.")
 	flags.StringVarP(&Importer.Index, "index", "i", "", "Pilosa index to import into.")
 	flags.StringVarP(&Importer.Frame, "frame", "f", "", "Frame to import into.")
-	flags.StringVarP(&Importer.Field, "field", "", "", "Field to import into.")
+	flags.StringVarP(&Importer.BSIGroup, "bsi-group", "", "", "BSIGroup to import into.")
```
Comment (jaffee) [inline cmd/import.go]: ugh, "bsi group" does not make for a very friendly term in the UI... we'll have to think of something to make this more approachable

File: ctl/import.go [hunk 3]
```diff
@@ -46,8 +46,8 @@ type ImportCommand struct {
 	// CreateSchema ensures the schema exists before import
 	CreateSchema bool
 
-	// For Range-Encoded fields, name of the Field to import into.
-	Field string `json:"field"`
+	// For Range-Encoded bsiGroups, name of the BSIGroup to import into.
```
Comment (jaffee) [inline ctl/import.go]: again, awk. Just making a note of these, I don't think we need to address them in this PR

File: cmd/import.go (hunk 2 above)
Comment (travisturner) [reply cmd/import.go]: well, i'm working to make `bsiGroup` un-exported, so we really shouldn't be using it anywhere externally.

but I agree, `bsiGroup` was not the really the best choice of terms given that it makes for awkward names and comments.

**Comment by travisturner** [comment]
@jaffee I agree. That's why this is WIP. I think step 2 (and one I think I should have done before this rename) is un-exporting it and removing anything `bsiGroup`-related from the external API. See #1345 for my proposal there.

**Comment by travisturner** [comment]
I opted to come at this from the other direction by un-exporting `Field` first. Renaming it to `bsiGroup` will come next. See #1351.

**Pull request closed without merging**